yaksh/views.py

```diff
@@ -313,15 +313,21 @@ def prof_manage(request):
         question_papers = QuestionPaper.objects.filter(quiz__course__creator=user,
                                                        quiz__is_trial=False
                                                        )
-        trial_quiz = Quiz.objects.filter(course__creator=user, is_trial=True)
+        trial_paper = AnswerPaper.objects.filter(user=user,
+                                                 question_paper__quiz__is_trial=True
+                                                 )
         if request.method == "POST":
             delete_quiz = request.POST.getlist('delete_quiz')
-            for quiz_id in delete_quiz:
-                quiz = Quiz.objects.get(id=quiz_id)
-                if quiz.course.is_trial == True:
-                    quiz.course.delete()
+            for answerpaper_id in delete_quiz:
+                answerpaper = AnswerPaper.objects.get(id=answerpaper_id)
+                if answerpaper.question_paper.quiz.course.is_trial == True:
+                    answerpaper.question_paper.quiz.course.delete()
                 else:
-                    quiz.delete()
+                    qpaper = answerpaper.question_paper
+                    if qpaper.answerpaper_set.count() == 1:
+                        answerpaper.question_paper.quiz.delete()
```

Comment:
You have already defined the variable qpaper, so you can use it instead of answerpaper.question_paper.
Actually before if statement itself you can define the variable and use it in pace of answerpaper.question_paper